### ISSUE TITLE
IP/Mask matching rules failure for socks5 upstream

### ISSUE THREAD

**Issue by davidblaisonneau-orange** [tinyproxy/tinyproxy#363]
Hi,

I would like to add socks proxy upstream to connect several remote services through a SSH socks proxy.

This works fine with the following config, setting the remote IP:
```
upstream socks5 socks_proxy:8080 "10.4.12.12"
upstream http corporate_proxy:8080
upstream none "localhost"
```

but not with a more generic one using a "IP/mask" filter
```
upstream socks5 socks_proxy:8080 "10.4.0.0/16"
upstream http corporate_proxy:8080
upstream none "localhost"
```
This config uses the default corporate proxy.

Is there anything i did not understood ?

**OS:**
- Debian 11 bullseye
- Tinyproxy 1.10.0-4

Thanks for your help !

**Comment by rofl0r** [comment]
please ask debian maintainers to update to 1.11.0 which has lots of bugfixes. i'm tired of getting bug reports for old versions. meanwhile you can compile tinyproxy 1.11.0 from source. the bug has been fixed in 23b0c8465306a0e069a639085bb08266b803a2de.

**Comment by davidblaisonneau-orange** [comment]
Thanks for your answer, and sorry bothering you.
